crates/ra_ide/src/syntax_highlighting.rs

```diff
@@ -539,21 +539,48 @@ fn highlight_element(
                 _ => h,
             }
         }
-        T![*] => {
-            let prefix_expr = element.parent().and_then(ast::PrefixExpr::cast)?;
-
-            let expr = prefix_expr.expr()?;
-            let ty = sema.type_of_expr(&expr)?;
-            if !ty.is_raw_ptr() {
-                return None;
-            } else {
-                HighlightTag::Operator | HighlightModifier::Unsafe
+        p if p.is_punct() => match p {
+            T![::] | T![->] | T![=>] | T![&] | T![..] | T![=] => HighlightTag::Operator.into(),
+            T![@] => HighlightTag::Operator | HighlightModifier::ControlFlow,
```

Comment:
I suggest sticking with just `Operator`. `ControlFlow` does not make sense semantically, and we should focus on what makes sense, rather than on what makes the current themes pretty. Themes should follows semantics rather than the other way round. Besides, the `@` operator is so exceedingly rare, that I doubt different coloring matter in practice.